README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

<!--badges: start -->

[![Travis build
status](https://travis-ci.com/debusklaneml/VirginiaC19.svg?branch=master)](https://travis-ci.com/debusklaneml/VirginiaC19)
[![CRAN\_Status\_Badge](https://www.r-pkg.org/badges/version/VirginiaC19)](https://cran.r-project.org/package=VirginiaC19)
[![lifecycle](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://www.tidyverse.org/lifecycle/#experimental)
[![License:
MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)
[![GitHub
commit](https://img.shields.io/github/last-commit/debusklaneml/VirginiaC19)](https://github.com/debusklaneml/VirginiaC19/commit/master)

<!--badges: end -->

## VirginiaC19

The VirginiaC19 package harvests the data from the [Virginia Department
of Health](https://www.vdh.virginia.gov/coronavirus/). This package is
currently a component to the [Covid19R
project](https://github.com/Covid19R).

Get the latest development version from
[github](https://github.com/debusklaneml/VirginiaC19)

``` r
devtools::install_github("debusklaneml/VirginiaC19")
```

## Getting the Most Up to Date Data

To get the most updated data, run the following functions

``` r
new_data <- refresh_VirginiaC19()
```

## Columns

The data follows the [covid19R standard for tidy Covid-19
data](https://covid19r.github.io/documentation/data-format-standard.html).


NEWS.md
# VirginiaC19 0.1.0

* Created first release
* Added a `NEWS.md` file to track changes to the package.


tests/testthat.R
library(testthat)
library(VirginiaC19)

test_check("VirginiaC19")


tests/testthat/test-refresh_VirginiaC19.R
refresh_col_names <-
  c("date",
    "location_code",
    "location",
    "vdh_health_district",
    "value",
    "deaths",
    "hospitalizations",
    "location_type",
    "data_type",
    "data_url",
    "data_set_name",
    "package_name",
    "function_to_get_data",
    "has_geospatial_info",
    "location_code_type"
  )

test_that("Check if columns still populate!", {
  
  res <- refresh_VirginiaC19()
  
  expect_named(res, refresh_col_names)
  
  expect_true(
    "Chesterfield" %in% unique(res$location)
  )  
})


setupcode.R
#' ----------------------------------------
#' Script to startup the a YOUR_PKG_NAME package
#' using usethis
#' ----------------------------------------

library(usethis)

# create_tidy_package(glue::glue"{here::here()/../PKGNAME})

# setup package
use_code_of_conduct()
use_data_raw()
# use_LICENSE_FUNCTION()

# imports packages
use_package("magrittr")
use_package("tibble")
#use_package("PKG_TO_IMPORT")

# suggests packages
#use_package("PKG_TO_SUGGEST", type = "Suggests")


# add other elements
use_news_md()
#use_vignette("VIGNETTE_FILENAME", "VIGNETTE TITLE")


#-----------------------------
# Cleanup before release ####
#-----------------------------

# tweaks to the description before release
use_tidy_description()
use_tidy_versions()

# Are there things you should cleanup in this package?
styler::style_dir("R")
#styler::style_dir("data-raw") #can comment out if not including local data
goodpractice::gp()

# make a pkgdown website!
pkgdown::build_site()

# for release - checks!
devtools::check_win_release()
devtools::release_checks()
devtools::spell_check()
devtools::check_rhub()

#-------------
# Release ####
#-------------
devtools::release()

R/refresh_VirginiaC19.R
#' Refresh the Virginia COVID-19 Data.
#'
#' @description Updates the Virginia COVID-19 cases.
#'
#' @details This package scrapes the state of Virginia reported cases.
#'
#' @param verbose Should messages be displayed?
#'
#' @source \href{https://github.com/debusklaneml/VirginiaC19}{VirginiaC19}
#' @source \href{https://github.com/Covid19R/documentation}{covid19R documentation}
#'
#' @return A tibble object
#' * date - The date in YYYY-MM-DD form
#' * location - The name of the location as provided by the data source.
#' * location_type - The type of location using the covid19R controlled vocabulary.
#' * location_code - A standardized location code using a national or international standard.
#' * location_code_type The type of standardized location code being used according to the covid19R controlled vocabulary.
#' * data_type - the type of data in that given row using the covid19R controlled vocabulary. Daily totals are found here.
#' * value - number of cases of each data type
#' @export
#'
#' @examples
#' \dontrun{
#' refresh_VirginiaC19()
#' }
#'
refresh_VirginiaC19 <- function(verbose = TRUE) {
  if (verbose) message(glue::glue("Downloading raw data from {url}."))

  read_data() %>%
    clean_data()
}


R/get_info_VirginiaC19.R
#' Get information about the dataset provided by VirginiaC19 package.
#'
#' @description Returns information about the datasets in this package for covid19R harvesting.
#'
#' @return a tibble of information about the datasets in this package.
#' @export get_info_VirginiaC19
#'
#' @examples
#' \dontrun{
#'
#' # get the dataset info from this package
#' get_info_VirginiaC19()
#' }
#'
#' @describeIn get_info_VirginiaC19 Basic information about the dataset.
get_info_VirginiaC19 <- function() {
  latest_data <-
    refresh_VirginiaC19(verbose = FALSE)

  dplyr::tibble(
    data_set_name = "VirginiaC19",
    package_name = "VirginiaC19",
    function_to_get_data = "refresh_VirginiaC19",
    data_details = "Open Source data from Virginia Department of Health. For more information, see https://www.vdh.virginia.gov/coronavirus/.",
    data_url = "https://www.vdh.virginia.gov/content/uploads/sites/182/2020/03/VDH-COVID-19-PublicUseDataset-Cases.csv",
    license_url = "https://github.com/debusklaneml/VirginiaC19/blob/master/LICENSE",
    data_types =
      latest_data %>%
        tidyr::drop_na(data_type) %>%
        dplyr::pull(data_type) %>%
        unique(),
    location_types =
      latest_data %>%
        tidyr::drop_na(location_type) %>%
        dplyr::pull(location_type) %>%
        unique(),
    spatial_extent = "county",
    has_geospatial_info = FALSE
  )
}


R/utils.R
#' Pipe operator
#'
#' See \code{magrittr::\link[magrittr:pipe]{\%>\%}} for details.
#'
#' @name %>%
#' @rdname pipe
#' @keywords internal
#' @export
#' @importFrom magrittr %>%
#' @usage lhs \%>\% rhs
NULL

url <- "https://www.vdh.virginia.gov/content/uploads/sites/182/2020/03/VDH-COVID-19-PublicUseDataset-Cases.csv"
# Original
# read_data <- function() {
#   suppressWarnings(
#     readr::read_csv(
#       url,
#       col_types =
#         readr::cols(
#           "Report Date" = readr::col_date(format = "%m/%d/%Y")
#         )
#     )
#  )
# }



# Trying different way.
read_data <- function() {
  suppressWarnings(
    readr::read_csv(url)
  )
}

# need to change the date into a date variable.
clean_data <- function(tbl) {
  suppressWarnings(
    tbl %>%
      janitor::clean_names() %>%
      dplyr::select(
        date = report_date,
        location_code = fips,
        location = locality,
        vdh_health_district,
        value = total_cases,
        deaths,
        hospitalizations
      ) %>%
      dplyr::mutate(
        location_type = "county",
        data_type = "cases_total",
        data_url = "https://www.vdh.virginia.gov/coronavirus/",
        data_set_name = "virginia_cases",
        package_name = "VirginiaC19",
        function_to_get_data = "refresh_VirginiaC19()",
        has_geospatial_info = FALSE,
        location_code_type = "fips_code"
      ) %>%
      dplyr::arrange(
        date
      ) %>%
      dplyr::mutate(date = lubridate::mdy(date))
  )
}
